sled-agent/src/services.rs

```diff
@@ -2760,6 +2762,18 @@ impl ServiceManager {
                             }
                             smfh.refresh()?;
                         }
+                        SwitchService::Pumpkind { asic } => {
+                            // The pumpkin daemon is only needed when running on
+                            // with real sidecar.
+                            if asic == &DendriteAsic::TofinoAsic {
```

Comment:
This has been tested in regular CI, on `a4x2`, and on real hardware, covering both cases for this `if`.